src/flutter/shell/platform/linux_embedded/window/native_window_drm_eglstream.cc

```diff
@@ -0,0 +1,299 @@
+    if (plane) {
+      drmModeFreePlane(plane);
+
+      constexpr char kPropNamePlaneType[] = "type";
+      auto type = GetPropertyValue(resources->planes[i], DRM_MODE_OBJECT_PLANE,
+                                   kPropNamePlaneType);
+      if (type == DRM_PLANE_TYPE_PRIMARY) {
+        return resources->planes[i];
+      }
+    }
+  }
+  // no plane found
+  return -1;
+}
+
+uint64_t NativeWindowDrmEglstream::GetPropertyValue(uint32_t id, uint32_t type,
+                                                    const char* prop_name) {
+  uint64_t value = -1;
+  auto properties = drmModeObjectGetProperties(drm_device_, id, type);
+  if (properties) {
+    for (uint32_t i = 0; i < properties->count_props; i++) {
+      auto property = drmModeGetProperty(drm_device_, properties->props[i]);
+      if (property) {
+        if (strcmp(prop_name, property->name) == 0) {
+          value = properties->prop_values[i];
+          drmModeFreeProperty(property);
+          break;
+        }
+        drmModeFreeProperty(property);
+      }
+    }
+    drmModeFreeObjectProperties(properties);
+  }
+  return value;
+}
+
+bool NativeWindowDrmEglstream::AssignAtomicRequest(drmModeAtomicReqPtr atomic) {
+  if (!CreatePropertyBlob() || !CreateFb()) {
+    return false;
+  }
+
+  struct drm_property_ids property_ids = {0};
+  GetPropertyIds(&property_ids);
+  drmModeAtomicAddProperty(atomic, drm_crtc_->crtc_id,
+                           property_ids.crtc.mode_id, drm_property_blob_);
+  drmModeAtomicAddProperty(atomic, drm_crtc_->crtc_id, property_ids.crtc.active,
+                           1);
+  drmModeAtomicAddProperty(atomic, drm_connector_id_,
+                           property_ids.connector.crtc_id, drm_crtc_->crtc_id);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.src_x, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.src_y, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.src_w,
+                           drm_mode_info_.hdisplay << 16);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.src_h,
+                           drm_mode_info_.vdisplay << 16);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_x, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_y, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_w,
+                           drm_mode_info_.hdisplay);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_h,
+                           drm_mode_info_.vdisplay);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.fb_id,
+                           drm_fb_);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_id,
+                           drm_crtc_->crtc_id);
+  return true;
+}
+
+void NativeWindowDrmEglstream::GetPropertyIds(
+    struct NativeWindowDrmEglstream::drm_property_ids* property_ids) {
```

Comment:
`struct NativeWindowDrmEglstream::drm_property_ids*` -> `NativeWindowDrmEglstream::drm_property_ids&`

Reply:
I fixed it.